crates/language/src/language.rs

```diff
@@ -414,11 +414,10 @@ impl SnippetTree {
             }
             // sanity check to avoid infinite loop
             if snippet_root.id() == id {
-                if snippet_root
+                if !snippet_root
                     .utf8_text(self.context.as_bytes())
                     .unwrap()
-                    .trim()
-                    != self.snippet.trim()
+                    .trim().contains(self.snippet.trim())
```

Comment:
when node has invisible tokens from external scanner, this won't be equal. (e.g automatic_semicolon in the php parser)

Reply:
I don't think this is right, many other languages including javascript which we by far have the most usage of have automatic semi-colon.

Reply:
You're right. After some debugging, I found out that `snippet_nodes_from_index` won't return a node if the biggest node isn't exactly equal to the snippet context. 
e.g `echo ^x` (using ^ as a prefix instead of $) will not return a snippet_node since it is not equal to `echo ^match;`
`echo ^match` should return a `Node(echo_statement, range(0,6))` however it returns nothing because `"echo ^x" != "echo ^x;`

This case usually applies to languages where semicolons are mandatory. E.g`echo ^x` should return a node in a language without semicolons, however in php you need to add the semicolon to get a Tree-sitter node `echo ^x;`

I had to change the tree-sitter parser for php to get this to work by changing it so that it returns a node without the semicolon. e.g I made this change to echo_statement
```js
//old echo_statement
echo_statement: $ => seq(
  keyword('echo'), $._expressions, $._semicolon,
),
//new echo_statement
echo_statement: $ => seq(
  keyword('echo'), $._expressions,
),
// and then I added echo_statement to a new node that has semicolon at the end 
no_semicolon: $ => seq(
  choice(
    $.echo_statement,
    $.exit_statement,
    .... //a bunch of other nodes
  ), 
  $._semicolon
),
```

If we could support partial node matches, it would make adding new languages easier in the future because we won't have to change the grammar file as much.

Reply:
ok, I reverted the grammar changes and added the semicolons back into the required snippets.